.github/workflows/sniff.yaml

```diff
@@ -0,0 +1,71 @@
+name: Sniff stage - quality control checks
+
+on:
+    pull_request:
+        branches: [master, develop]
+    push:
+        branches: [master]
+
+jobs:
+    qa_checks:
+        name: Quality control checks
+        runs-on: ubuntu-latest
+        env:
+            PHPCS_BRANCH: "dev-master"
+            WPCS_BRANCH: "dev-master"
+
+        steps:
+            # Checkout repository
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            # Setup PHP versions, run checks
+            - name: PHP setup
+              uses: shivammathur/setup-php@v2
+              with:
+                php-version: '7.4'
+                tools: composer:v1
+              env:
+                COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Install xmllint
+              run: sudo apt-get install libxml2-utils
+
+            - name: Get composer cache directory
+              id: composer-cache
+              run: |
+                echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Cache composer dependencies
+              uses: actions/cache@v1
+              with:
+                path: ${{ steps.composer-cache.outputs.dir }}
+                key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+                restore-keys: |
+                  ${{ runner.os }}-composer-
+
+            - name: Set the required PHPCS and WPCS versions
+              run: composer require squizlabs/php_codesniffer:${PHPCS_BRANCH} wp-coding-standards/wpcs:${WPCS_BRANCH} --no-update --no-scripts
+
+            - name: Install the rest of the dependencies
+              run: composer install
+
+            - name: Check the code style of the WPThemeReview codebase
+              run: composer check-cs
+
+            - name: Lint the PHP files against parse errors
+              run: if find . -path ./vendor -prune -o -name "*.php" -exec php -l {} \; | grep "^[Parse error|Fatal error]"; then exit 1; fi
+
+            - name: Validate the composer.json file.
+              run: composer validate --no-check-all --strict
+
+            - name: Validate xml files
+              run: xmllint --noout --schema ./vendor/squizlabs/php_codesniffer/phpcs.xsd ./*/ruleset.xml
+
+            - name: Check the code-style consistency of the xml files.
+              run: |
+                export XMLLINT_INDENT=$'\t'
```

Comment:
Any reason not to just define this as a global environment variable instead?

Reply:
Just because I wasn't aware that was an option (saw it in your WPCS PR) 😅 

I will definitely move it to a global env var 👍🏼